polkadot/node/overseer/src/lib.rs

```diff
@@ -82,6 +82,7 @@ pub use polkadot_node_subsystem_types::{
 };
 
 use subspace_runtime_primitives::{opaque::Block, BlockNumber, Hash};
+use sc_consensus_subspace::{notification::SubspaceNotificationStream, NewSlotInfo, NewSlotNotification};
```

Comment:
Honestly, I'm not yet sure about what the exact pieces are needed, can be done once we are clear about that.

Reply:
I'd really like to deal with it early instead because this required a whole bunch of packages to pull `sc-consensus-subspace` that already made dependencies messy. If we continue that way in a few iterations you'll not be able to figure out easily why dependencies are added without loading the whole thing into the head, which is very hard to do.

Reply:
Should be addressed in https://github.com/subspace/subspace/pull/196/commits/f8d023844096e733561434c406945f9957ffcf0b. The last thing I'd like to tackle in this PR is to remove the deprecated candidate receipt code.

BTW, now `cirrus-node-primitives` is also used by `subspace-node`.

Reply:
Think it over again, I'll defer the candidate receipt removal to another PR so that this PR is all about the new features. :(